### PR TITLE
Clarify claims about PhantomData<T>.

### LINKED ISSUE #27523: docs: clarify `PhantomData` vs "phantom types" statement
In the documentation for [`PhantomData`](https://doc.rust-lang.org/stable/core/marker/struct.PhantomData.html), it states that: "Though they both have scary names, PhantomData<T> and "phantom types" are unrelated."

This seems to not be an accurate description of the relationship between the PhantomData marker and phantom types, the concept. They are related, just not the same thing.

/cc @steveklabnik

### PR DESCRIPTION
**Pull request by steveklabnik**
This wording was too strong.

Fixes #27523

### CODE REVIEW

**Comment by Gankro** [comment]
This just seems more confusing to me

**Comment by Gankro** [comment]
This seems like one of those "you gotta let it go" kinda sentences. Where there's something you want to say but it's not worth trying to do it unless you want to commit to a full explanation.

**Comment by steveklabnik** [comment]
So maybe @nikomatsakis or @aturon can weigh in here. IIRC, the relationship is that you can use PhantomData to implement phantom types, but not every usage is such.

**Comment by steveklabnik** [comment]
(I am willing to commit to a full explanation, it's just that I'm only two pages into TAPL :( )

**Comment by aturon** [comment]
@steveklabnik Yeah, that's the correct relationship. But I agree with @Gankro that this is getting into the weeds.

**Comment by Gankro** [comment]
fwiw I describe PhantomData here: https://doc.rust-lang.org/nightly/adv-book/phantom-data.html

**Comment by steveklabnik** [comment]
We originally added this sentence because people were asking about the relationship, and/or there were old blog posts talking about phantom types where you'd upgrade the code to PhantomData. Hm.

**Comment by Gankro** [comment]
Phantom types is a pattern, and strictly speaking doesn't need PhantomData in the usual case. For instance, 

```rust
struct Sanitized;
struct Raw;

struct String<Sanitization> {
    data: String,
    _sanitization: Sanitization,
}
```

```rust
struct Sanitized;
struct Raw;

struct String<Sanitization> {
    data: String,
    _sanitization: PhantomData<Sanitization>,
}
```

Are semantically identical, differing only in how they're initialized.

**Comment by steveklabnik** [comment]
@gankro try this wording on for size

File: src/libcore/marker.rs [hunk 1]
```diff
@@ -273,7 +273,11 @@ macro_rules! impls{
 /// even though it does not. This allows you to inform the compiler about certain safety properties
 /// of your code.
 ///
-/// Though they both have scary names, `PhantomData<T>` and "phantom types" are unrelated. 👻👻👻
+/// # A ghastly note 👻👻👻
```
Comment (Gankro) [inline src/libcore/marker.rs]: YES

File: src/libcore/marker.rs [hunk 2]
```diff
@@ -273,7 +273,11 @@ macro_rules! impls{
...
+///
+/// Though they both have scary names, `PhantomData<T>` and 'phantom types' are related, but not
+/// identical. Phantom types are a more general concept that doesn't require `PhantomData<T>` to
```
Comment (Gankro) [inline src/libcore/marker.rs]: s/doesn't/don't?

**Comment by Gankro** [comment]
r=me with nit

File: src/libcore/marker.rs (hunk 2 above)
Comment (steveklabnik) [inline src/libcore/marker.rs]: this is a tricky english thing. I forget the rule here, it depends if "Phantom types" or "concept" is the subject. I think you're right.

**Pull request merged**